core/src/main/scala/akka/kafka/ProducerSettings.scala

```diff
@@ -146,7 +148,8 @@ final class ProducerSettings[K, V](
     val valueSerializerOpt: Option[Serializer[V]],
     val closeTimeout: FiniteDuration,
     val parallelism: Int,
-    val dispatcher: String
+    val dispatcher: String,
+    val eosCommitIntervalMs: Long
```

Comment:
I don't know if that makes sense but in some use case the user might want to be able to decide to commit a transaction after a specific number of input message were fully processed.
I think it should be doable by tracking an `awaitingTransactionCommit` the same way the current `awaitingConfirmation` works.
This way you could support the use case where the user want to make sure each input message generate a transaction (or more than one if wanted)

Reply:
That's a fair point.  There are several reasons I didn't implement this approach, but they may not be too compelling.  I welcome a challenge here.

1. [KIP-129](https://cwiki.apache.org/confluence/display/KAFKA/KIP-129%3A+Streams+Exactly-Once+Semantics) nor the current Kafka Streams implementation (AFAICT) exposes a way to batch a transaction in this manner.
2. In a Consume-Transform-Produce workflow we have strong guarantees that the source message is already persisted and possibly replicated, so it's not critical we must process the message successfully the first time.  If the batch fails for some transient reason then it can be processed the next time.  I acknowledge this reason is more of an opinion than fact since there are other things that can go wrong, such as the retention period for that source message elapsing, etc.
3. It would be a little more implementation work and I wanted to see what people thought of this approach (which mirrors what's done in Kafka Streams) before I extended the implementation.

Reply:
Your points are completely valid.
Let's get this implementation merged and after I could take a look at adding this feature if I really feel it important !

Reply:
Also think we need extra support on top of this current Kafka transactions semantics to implement things like `sagas`, especially in the context of "data pipelines" and not just "microservices".

https://microservices.io/patterns/data/saga.html

cc @seglo @fchaillou